src/main/java/com/thealgorithms/datastructures/trees/BinaryTree.java

```diff
@@ -150,80 +150,78 @@ else if (temp.parent.data < temp.data) {
             } else {
                 temp.parent.left = null;
             }
-            return true;
         } // Two children
-        else if (temp.left != null && temp.right != null) {
-            Node successor = findSuccessor(temp);
-
-            // The left tree of temp is made the left tree of the successor
-            successor.left = temp.left;
-            successor.left.parent = successor;
-
-            // If the successor has a right child, the child's grandparent is it's new parent
-            if (successor.parent != temp) {
-                if (successor.right != null) {
-                    successor.right.parent = successor.parent;
-                    successor.parent.left = successor.right;
-                    successor.right = temp.right;
-                    successor.right.parent = successor;
-                } else {
-                    successor.parent.left = null;
+        else {
```

Comment:
I wouldn't do that. The logic behind `if - else if - else` structure is more readable and easier to understand.